data/projects/debug_profile.md

```diff
@@ -1,26 +1,50 @@
-In the development lifecycle of a Bear it needs to be debugged and profiled
-and coala should provide a comprehensive environment
-for this. In this project the participant shall develop a system to easily
+A famous programming quote reads "Debugging is twice as hard as writing code
+in the first place". Thus even though writing code can be as trivial as
+opening up your favorite text editor and crunching some keys we need tools to
+help us debug our code. An essential toolset that coala lacks is the ability
+to debug its Bears, i.e. the analysis procedures. In the development lifecycle
+of a Bear it needs to be debugged and profiled and coala should provide a
+comprehensive environment for this.
+
+In this project the participant shall develop a system to easily
 step into the analysis code (likely using the Pdb module already integrated
-into Python) and step out as soon as the analysis finished.
+into Python) and step out as soon as the analysis finished. The following are desireable goals for such a toolset's features:
+
+1. The ability to jump directly into Bear code and move through it using a
+debugger interface such as pdb.
+2. The toolset should also provide the ability to invesitgate the settings
+passed to the Bear.
+3. Inspect the results created by the Bear.
+
+This rich source of information allows Bear developers to create better
+performant Bears through the profiling interface. It also makes Bear
+developement easier through the debugging interface. Also, it gives the added
+benefit of better quality Bears being contributed to the community.
 
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
 * The applicant has made a [cEP](https://coala.io/cep) of how the profiling
   will be implemented.
 * The applicant is familiar with pdb.
 * The applicant is familiar with core coala codebase.
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
 * Pdb should be integrated completely into the coala core.
+* Documentation and tests for the above task should also have been completed.
+
+##### CODING PHASE 2
+
 * Settings of Bears can be accessed in the debug environment.
 * Documentation and tests for the above task should also have been completed.
+* The profile option to CLI is added.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 3
 
-* The profile option to CLI is added and merged.
-* The ability to inspect results produced by the bear is implemented.
+* The profile option to CLI is merged.
+* The ability to inspect results produced by the Bear is implemented.
```

Comment:
the bear -> Bears . No specific bear to implement this in was mentioned but rather we want this wo work in all bears right?